### DISCUSSION TITLE
Canonical's wishes & priorities for .NET 2024

### DISCUSSION THREAD

**Discussion by dviererbe** [dotnet/source-build#4072] [General]
We at Canonical have been thinking too, about our wishes and priorities for the .NET Project in 2024. The list is roughly sorted in order of priority/importance:

# Security fixes as separate patches
Because .NET LTS packages are part of Ubuntu LTS releases, they will receive 5 years of standard security maintenance and 10 years of expanded security maintenance (ESM) with Ubuntu Pro (see https://ubuntu.com/about/release-cycle for further information).

This exceeds the [support period provided by Microsoft](https://dotnet.microsoft.com/en-us/platform/support/policy/dotnet-core). Canonical needs to build the skill and capacity to be able to provide the longer support timeframe. Canonical is willing to do that, but expects some help from Microsoft in terms of how the security fixes for .NET versions supported by Microsoft are shared with partners.

If Microsoft could share the security fixes for .NET versions supported by Microsoft as separately identifiable patches with mappings to the CVE they fix, Canonical can use those to minimize the effort of backporting the fixes onto the EoL .NET versions that we still need to support.

# Frictionless user experience for .NET Snaps

The current user experience with .NET Snaps is suboptimal. Canonical is committed to takeover the .NET Snaps currently published by Microsoft to maintain and improve these.

# Improve transition experience from Microsoft package feed to Ubuntu archive debs

A common issue with .NET users on Ubuntu is that they initially instal .NET from the Microsoft package feed and later want to switch to packages from the Ubuntu archive which often results in a conflict, because apt prioritizes the packages in the Microsoft feed.

Unfortunately many of these users are not aware of the excellent help pages by Microsoft for these kind of issues (e.g. https://learn.microsoft.com/en-us/dotnet/core/install/linux-package-mixup?pivots=os-linux-ubuntu#solutions), get frustrated and may receive suboptimal advice by other community members.

We would like to investigate if Canonical or Microsoft can do any changes to reduce the likelihood for these kinds of issues.

# Distro customizations

The .NET CLI provides help links in various situations, for example:

- part of the `dotnet info` output is:
  ```
  Download .NET:
	https://aka.ms/dotnet/download
  ```
- part of the output when trying to launch a .NET 6 compiled `.dll` with .NET 8:
  ```
  To install missing framework, download:
  https://aka.ms/dotnet-core-applaunch?framework=Microsoft.NETCore.App&framework_version=6.0.0&arch=x64&rid=ubuntu.23.10-x64&os=ubuntu.23.10
  ```

We would like to include or even replace these links with distro specific advice (in case of Ubuntu, that `sudo apt install dotnetX` could be used).

**Comment by MichaelSimons** [comment]
Regarding `Improve transition experience from Microsoft package feed to Ubuntu archive debs` - mentioning another usage scenario in which this comes up is when users need to register the Microsoft package feed to install non-dotnet packages (e.g. other microsoft package that are not in the native package feed) but would prefer to prioritize the native package feeds for the dotnet packages.  The guidelines [help with this](https://learn.microsoft.com/en-Us/dotnet/core/install/linux-package-mixup?pivots=os-linux-ubuntu#my-linux-distribution-provides-net-packages-and-i-want-to-use-them) but often times users get hung up.
